src/workerd/api/global-scope.c++

```diff
@@ -470,6 +471,21 @@ void ServiceWorkerGlobalScope::sendHibernatableWebSocketMessage(
   KJ_IF_SOME(h, exportedHandler) {
     KJ_IF_SOME(handler, h.webSocketMessage) {
       auto promise = handler(lock, kj::mv(websocket), kj::mv(message));
+
+      // If we have a maximum event duration timeout set, we should prevent the actor from running
+      // for more than the user selected duration.
+      KJ_IF_SOME(timeout, eventTimeout) {
+        if (timeout > 0) {
+          // We must have an actor if we are delivering an hibernatable web socket message.
+          auto& actor = KJ_ASSERT_NONNULL(IoContext::current().getActor());
+
+          promise = promise.exclusiveJoin(IoContext::current().afterLimitTimeout(timeout * kj::SECONDS)
```

Comment:
Why did we change from miliseconds to seconds? It seems weird to decrease granularity here?